File: java/adaptersdk-schema-lib/src/main/java/io/github/fileanalysissuite/adaptersdk/schema/FieldImpl.java
/*
 * Copyright 2022 Micro Focus or one of its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.github.fileanalysissuite.adaptersdk.schema;

import javax.annotation.Nonnull;
import io.github.fileanalysissuite.adaptersdk.schema.model.Field;

class FieldImpl implements Field
{
    private final String fieldName;
    private final String fieldType;
    private final ObjectEncoding objectEncoding;
    private final boolean multivalue;
    private final boolean mandatory;
    private final boolean caseInsensitive;
    private final boolean tokenizedPath;


    FieldImpl(
        final String fieldName,
        final String fieldType,
        final ObjectEncoding objectEncoding,
        final boolean multivalue,
        final boolean mandatory,
        final boolean caseInsensitive,
        final boolean tokenizedPath
        )
    {
        this.fieldName = fieldName;
        this.fieldType = fieldType;
        this.objectEncoding = objectEncoding;
        this.multivalue = multivalue;
        this.mandatory = mandatory;
        this.caseInsensitive = caseInsensitive;
        this.tokenizedPath = tokenizedPath;
    }

    @Nonnull
    @Override
    public String getFieldName()
    {
        return fieldName;
    }

    @Override
    public String getFieldType()
    {
        return fieldType;
    }


    @Override
    public ObjectEncoding getObjectEncoding()
    {
        return objectEncoding;
    }

    @Override
    public Field getParentField()
    {
        return null;
    }

    @Override
    public boolean isMultivalue()
    {
        return multivalue;
    }

    @Override
    public boolean isMandatory()
    {
        return mandatory;
    }

    @Override
    public boolean isCaseInsensitive()
    {
        return caseInsensitive;
    }

    @Override
    public boolean isTokenizedPath()
    {
        return tokenizedPath;
    }

}

File: dotnet/src/OpenText.FAS.AdapterSdkSchema/FieldImpl.cs
/**
 * Copyright 2022-2024 Open Text.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
namespace OpenText.FAS.AdapterSdkSchema
{
    internal class FieldImpl : IField
    {
        private readonly string fieldName;
        private readonly string fieldType;
        private readonly ObjectEncoding? objectEncoding;
        private readonly bool multivalue;
        private readonly bool mandatory;
        private readonly bool caseInsensitive;
        private readonly bool tokenizedPath;

        public FieldImpl(
            string fieldName,
            string fieldType,
            ObjectEncoding? objectEncoding,
            bool multivalue,
            bool mandatory,
            bool caseInsensitive,
            bool tokenizedPath
        )
        {
            this.fieldName = fieldName;
            this.fieldType = fieldType;
            this.objectEncoding = objectEncoding;
            this.multivalue = multivalue;
            this.mandatory = mandatory;
            this.caseInsensitive = caseInsensitive;
            this.tokenizedPath = tokenizedPath;
        }

        public string FieldName => fieldName;

        public string FieldType => fieldType;

        public ObjectEncoding? ObjectEncoding => objectEncoding;

        public virtual IField ParentField => null;

        public bool IsMultivalue => multivalue;

        public bool IsMandatory => mandatory;

        public bool IsCaseInsensitive => caseInsensitive;

        public bool IsTokenizedPath => tokenizedPath;
    }
}
